src/pages/Home/components/SearchBar.tsx

```diff
@@ -1,15 +1,29 @@
-import { useNavigate } from 'react-router-dom';
+import { useNavigate, useSearchParams } from 'react-router-dom';
 import SearchIconGray from '@/assets/Main/search-icon-gray.svg';
+import { getCategoryNameById } from '@/constants/shopCategories';
+import useLogger from '@/util/hooks/analytics/useLogger';
 
 export default function SearchBar() {
+  const logger = useLogger();
   const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
 
+  const handleSearchBarClick = () => {
+    const categoryParam = searchParams.get('category');
+    const categoryName = getCategoryNameById(categoryParam ? Number(categoryParam) : undefined);
```

Comment:
해당 식이 반복되는 것 같은데 categoryId가 0이 될 일이 없다면 아래처럼 분리해줘도 가독성이 좋을 것 같네요!

```suggestion
    const categoryId = categoryParam && Number(categoryParam);
    const categoryName = getCategoryNameById(categoryId);
```

Reply:
[61e28c1](https://github.com/BCSDLab/KOIN_ORDER_WEBVIEW/pull/214/commits/61e28c13f61ab5f9a6f5768f363220195adaa9e5)

getCategoryNameById의 타입 조건때문에 위처럼 수정했는데 괜찮을까요?

Reply:
undefined를 가질 수도 있는게 문제가 되긴 하네요.. 우선 이렇게 넘어갑시다..!